app/derivative_services/audio_derivative_service.rb

```diff
@@ -38,17 +38,54 @@ def valid?
   end
 
   def create_derivatives
-    _stdout, _stderr, status = Open3.capture3("ffmpeg", "-y", "-i", file_object.disk_path.to_s, "-codec:a", "libmp3lame", temporary_output.path.to_s)
-    return unless status.success?
-    change_set.files = [build_file]
+    Dir.mktmpdir do |dir|
+      dir = Pathname.new(dir)
+      output, built_files = generate_hls_derivatives(dir)
+      break unless built_files
+      generate_hls_playlist(output, dir, built_files)
+    end
+  end
+
+  def generate_hls_playlist(output, dir, built_files)
+    hls_file = dir.join("hls.m3u8")
+    content = File.read(hls_file)
+    built_files.each do |file, id|
+      content.gsub!(file, helper.download_url(resource.id, id))
+    end
+    File.open(hls_file, "w") do |f|
+      f.puts content
+    end
+    change_set = DynamicChangeSet.new(output).prepopulate!
+    change_set.files = [build_file(hls_file, filename: "hls.m3u8")]
     change_set_persister.buffer_into_index do |buffered_persister|
       buffered_persister.save(change_set: change_set)
     end
   end
 
+  def generate_hls_derivatives(dir)
+    _stdout, _stderr, status =
+      Open3.capture3("ffmpeg", "-y", "-i", file_object.disk_path.to_s, "-hls_list_size", "0", "-hls_time", "10", "-f", "hls", "-codec:a", "libmp3lame", dir.join("hls.m3u8").to_s)
```

Comment:
Do we want to move any of these settings to the config file like we do for kakadu?

Reply:
I don't think that would improve readability, personally. Seems easier to have the command here.

Reply:
I'm fine with that, and I don't remember the thinking behind putting the jp2 recipes in the config file instead of in code like this.

Reply:
It was just because that's how Hydra::Derivatives did it.